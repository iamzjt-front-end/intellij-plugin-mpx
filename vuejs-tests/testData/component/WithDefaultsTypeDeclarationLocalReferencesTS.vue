<script setup lang="ts">
const defaults = {
  prop: 1,
};
withDefaults(defineProps<{
  prop: number | undefined,
}>(), {...defaults}); // not supported by Vue, ignore

prop; // TODO Should be unresolved
</script>

<template>
  {{prop}}
</template>
