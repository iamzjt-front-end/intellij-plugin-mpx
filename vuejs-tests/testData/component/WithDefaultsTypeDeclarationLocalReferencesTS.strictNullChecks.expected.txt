{
 class: "VueSourceComponent",
 components: {
  WithDefaultsTypeDeclarationLocalReferencesTS: <recursive value of class VueSourceComponent>,
 },
 props: [
  {
   class: "VueScriptSetupInputProperty",
   name: "prop",
   jsType: "number | undefined",
   required: true,
  },
 ],
}
